Fix db model for CNV

dashaylan opened the pull request:
This PR changes the CNV model and all files referencing the model to refer to chromosome_number as chromosome. It also sets the PK of the CNV table to include the chromosome, creating a composite key composed of the sample_id, start_position and chromosome.

dashaylan committed:
Add initial auth methods.

Rather than storing a public key, the access method will have to communicate with keycloak to obtain it.

dashaylan committed:
Update spec with sample object and access_level (#21)

dashaylan committed:
Add keycloak handler

dashaylan committed:
Change chromosome_number to chromosome, add to CNV PK

dashaylan committed:
Remove auth WIP

dashaylan committed:
Fix chromosome spelling

dashaylan commented:
Since this PR is focused on model changes, should Sample remain as it is, or should we change it to Dataset to keep some sort of consistency among the services? Both Jimmy and I have gotten a bit confused over this already.

dashaylan committed:
Remove test data structs

haoyuanli commented:
I agree, I think it's important to get clarifications here

zbozoky commented:
Let's discuss it tomorrow, because I am not sure I hundred percent understand what you meant .

dashaylan committed:
Change 'Patient' to 'Dataset'

Spec updated to use Dataset as the base path rather than patient. Routes, ORM and test code updated to reflect new spec.
